data/repository/src/commonMain/kotlin/social/androiddev/common/repository/timeline/TimelineRepoModule.kt

```diff
@@ -0,0 +1,125 @@
+ *
+ * You should have received a copy of the GNU General Public License along with Dodo. If not, see <https://www.gnu.org/licenses/>.
+ */
+package social.androiddev.common.repository.timeline
+
+import com.squareup.sqldelight.runtime.coroutines.asFlow
+import com.squareup.sqldelight.runtime.coroutines.mapToList
+import kotlinx.coroutines.flow.map
+import org.koin.core.module.Module
+import org.koin.dsl.module
+import org.mobilenativefoundation.store.store5.Bookkeeper
+import org.mobilenativefoundation.store.store5.Market
+import org.mobilenativefoundation.store.store5.NetworkFetcher
+import org.mobilenativefoundation.store.store5.NetworkUpdater
+import org.mobilenativefoundation.store.store5.OnNetworkCompletion
+import org.mobilenativefoundation.store.store5.Store
+import social.androiddev.common.network.MastodonApi
+import social.androiddev.common.network.model.Status
+import social.androiddev.common.persistence.localstorage.DodoAuthStorage
+import social.androiddev.common.timeline.TimelineDatabase
+import social.androiddev.common.timeline.TimelineItem
+
+/**
+ * Koin module containing all koin/bean definitions for
+ * timeline repository delegates.
+ */
+val timelineRepoModule: Module = module {
+
+    factory<HomeTimelineRepository> { RealHomeTimelineRepository(get()) }
+
+    factory<Store<FeedType, List<TimelineItem>, List<TimelineItem>>> {
+        val database = get<TimelineDatabase>()
+        Store.by(
+            reader = { key: FeedType ->
+                when(key){
+                    is FeedType.Home ->  get<TimelineDatabase>()
+                        .timelineQueries
+                        .selectHomeItems()
+                        .asFlow()
+                        .mapToList().map {
+                            it.ifEmpty { return@map null }
+                        }
+                }
+
+            },
+            writer = { _, input ->
+                input.forEach(database::tryWriteItem)
+                true
+            },
+            deleter = { TODO() },
+            clearer = { TODO() }
+        )
+    }
+    factory {
+        //Todo Add logic for conflict resolution handling for when we start posting toots
+        Bookkeeper.by(
+            read = { _: FeedType -> null },
+            write = { _, _ -> true },
+            delete = { TODO() },
+            deleteAll = { TODO() }
+        )
+    }
+
+    factory {
+        NetworkFetcher.by(
+            get = { key: FeedType ->
+                when(key) {
+                    is FeedType.Home -> {
+                        val authStorage = get<DodoAuthStorage>()
+                        get<MastodonApi>()
+                            .getHomeFeed(authStorage.currentDomain!!, authStorage.getAccessToken(authStorage.currentDomain!!)!!)
```

Comment:
`!!` 😭  We'll fix this later

Reply:
ha yeah, really I'd love a flow<Pair<Domain, Token>() that I can switchmap from. This way if the user changes we automatically emit a new feed